### PR TITLE
Send back to referrer (if local) on a lot of forms on the submission pages. Fixes #1355.

### LINKED ISSUE #1355: Send back to previous page instead of defined page on some forms
For example when claiming or verifying a problem you get send back to the submission page. Better would be to send back to the referrer

### PR DESCRIPTION
**Pull request by nickygerritsen**

### CODE REVIEW

File: webapp/src/Controller/Jury/SubmissionController.php [hunk 1]
```diff
@@ -1102,7 +1106,7 @@ public function verifyAction(
             $redirect = $this->generateUrl('jury_submission_by_judging', ['jid' => $judgingId]);
         }
 
-        return $this->redirect($redirect);
+        return $this->redirectToLocalReferrer($this->router, $request, $redirect);
```
Comment (thijskh) [inline webapp/src/Controller/Jury/SubmissionController.php:1109]: Old line 1096/new line 1100 above has a comment that describes the logic that follows, i think with this change the described logic/distinction is undone?

File: webapp/src/Controller/Jury/SubmissionController.php [hunk 2]
```diff
@@ -1213,9 +1217,13 @@ protected function processClaim($judging, Request $request, ?string &$claimWarni
...
                 if ($action === 'claim') {
-                    return $this->redirectToRoute('jury_submission', ['submitId' => $judging->getSubmission()->getSubmitid()]);
+                    return $this->redirectToLocalReferrer($this->router, $request,
+                        $this->generateUrl('jury_submission', ['submitId' => $judging->getSubmission()->getSubmitid()])
+                    );
                 } else {
-                    return $this->redirectToRoute('jury_submissions');
+                    return $this->redirectToLocalReferrer($this->router, $request,
+                        $this->generateUrl('jury_submissions')
+                    );
```
Comment (thijskh) [inline webapp/src/Controller/Jury/SubmissionController.php:1226]: Maybe (?) symmetrically, if you unclaim something you're done with it and will return to the list

File: webapp/src/Controller/Jury/SubmissionController.php [hunk 3]
```diff
@@ -1136,7 +1140,7 @@ public function verifyShadowDifferenceAction(
             $redirect = $this->generateUrl('jury_submission_by_external_judgement', ['externalJudgement' => $extjudgementid]);
         }
 
-        return $this->redirect($redirect);
+        return $this->redirectToLocalReferrer($this->router, $request, $redirect);
```
[the remark this answers is not in the document]
Comment (nickygerritsen) [reply webapp/src/Controller/Jury/SubmissionController.php:1143]: I see the comment but can't seem to find where we set the `redirect` var. I think in this case it's better to always redirect to the referrer if possible.

File: webapp/src/Controller/Jury/SubmissionController.php (hunk 1 above)
Comment (nickygerritsen) [reply webapp/src/Controller/Jury/SubmissionController.php:1109]: Same here

File: webapp/src/Controller/Jury/SubmissionController.php (hunk 2 above)
Comment (nickygerritsen) [reply webapp/src/Controller/Jury/SubmissionController.php:1226]: Discussed with @eldering and @meisterT and we think that is not better. So I will leave this part as this PR does it.

File: webapp/src/Controller/Jury/SubmissionController.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (thijskh) [reply webapp/src/Controller/Jury/SubmissionController.php:1143]: Yes, so it's better now to update the comment and remove the now definitely unused code that dealt with it.

**Review by thijskh — APPROVED** [review]

**Pull request merged**